File: src/main/java/io/github/onepieceapi/controllers/CharacterController.java
package io.github.onepieceapi.controllers;

import static io.github.onepieceapi.constants.Urls.ALL_URL;
import static io.github.onepieceapi.constants.Urls.CHARACTER_URL;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import io.github.onepieceapi.model.Character;
import io.github.onepieceapi.services.CharacterService;
import lombok.AllArgsConstructor;

@RestController
@RequestMapping(CHARACTER_URL)
@AllArgsConstructor
public class CharacterController {
    private final CharacterService characterService;

    @GetMapping(ALL_URL)
    public ResponseEntity<Page<Character>> getCharacter(Pageable pageable) {
        return ResponseEntity.ok(characterService.getAll(pageable));
    }
}

File: src/main/kotlin/io/github/onePieceApi/controllers/CharacterController.kt
package io.github.onePieceApi.controllers

import io.github.onePieceApi.constants.Urls.ALL_URL
import io.github.onePieceApi.constants.Urls.CHARACTER_URL
import io.github.onePieceApi.model.Character
import io.github.onePieceApi.services.CharacterService
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(CHARACTER_URL)
class CharacterController constructor(
    private val characterService: CharacterService
) {

    @GetMapping(ALL_URL)
    fun getCharacter(pageable: Pageable): ResponseEntity<Page<Character>> {
        val result = characterService.getAll(pageable)
        return ResponseEntity.ok(result)
    }
}
